RevenueCat/Plugins/Android/PurchasesWrapper.java

```diff
@@ -0,0 +1,331 @@
+package com.revenuecat.purchasesunity;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.revenuecat.purchases.PurchaserInfo;
+import com.revenuecat.purchases.Purchases;
+import com.revenuecat.purchases.PlatformInfo;
+import com.revenuecat.purchases.common.CommonKt;
+import com.revenuecat.purchases.common.ErrorContainer;
+import com.revenuecat.purchases.common.MappersKt;
+import com.revenuecat.purchases.common.OnResult;
+import com.revenuecat.purchases.common.OnResultList;
+import com.revenuecat.purchases.interfaces.UpdatedPurchaserInfoListener;
+import com.unity3d.player.UnityPlayer;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class PurchasesWrapper {
+    private static final String RECEIVE_PRODUCTS = "_receiveProducts";
+    private static final String GET_PURCHASER_INFO = "_getPurchaserInfo";
+    private static final String MAKE_PURCHASE = "_makePurchase";
+    private static final String CREATE_ALIAS = "_createAlias";
+    private static final String RECEIVE_PURCHASER_INFO = "_receivePurchaserInfo";
+    private static final String RESTORE_TRANSACTIONS = "_restoreTransactions";
+    private static final String IDENTIFY = "_identify";
+    private static final String RESET = "_reset";
+    private static final String GET_OFFERINGS = "_getOfferings";
+    private static final String CHECK_ELIGIBILITY = "_checkTrialOrIntroductoryPriceEligibility";
+
+    private static final String PLATFORM_NAME = "unity";
+    private static final String PLUGIN_VERSION = "2.1.0-SNAPSHOT";
+
+    private static String gameObject;
+    private static UpdatedPurchaserInfoListener listener = new UpdatedPurchaserInfoListener() {
+        @Override
+        public void onReceived(@NonNull PurchaserInfo purchaserInfo) {
+            sendPurchaserInfo(MappersKt.map(purchaserInfo), RECEIVE_PURCHASER_INFO);
+        }
+    };
+
+    public static void setup(String apiKey,
+                             String appUserId,
+                             String gameObject_,
+                             boolean observerMode,
+                             String userDefaultsSuiteName) {
```

Comment:
this last param is new, it's just ignored. I debated whether to even pass it all the way here, but I figured it was better for the actual java layer to determine what it needs or what it doesn't, since the intermediate layers should be as dumb as possible.